### ISSUE TITLE
[Suggestion] Split allUnits array into side arrays

### ISSUE THREAD

**Issue by wurzel0701** [official-antistasi-community/A3-Antistasi-1.4#152]
It would be better to split the allUnits array into several smaller ones, sorted by the units side (as the markers are). This would improve runtime on every check if enemies are near operation, as the number of units to search is smaller. The allUnits array can be kept for all operations needing every side. 

Would not be such a big chance, propably not even a great optimisation, but should save a few ms on some scripts

**Comment by Bob-Murphy** [comment]
If I understand this correctly this, this suggestion only targets performance, right?

**Comment by wurzel0701** [comment]
Mostly yes, it would make it a bit easier for devs too, but thats insignificant

**Comment by PoweredByPot** [comment]
used 134 times in 52

**Comment by wurzel0701** [comment]
I would suggest splitting this whole array up into multiple arrays, like

```sqf
allPlayersRebels
allPlayersOccupants 
allPlayersInvaders
allAIRebels
allAIOccupants
allAIInvaders
```
and then adding two functions similar to distanceUnits. with the following behaviour:

```sqf
[position or object, radius, side, mode] call A3A_fnc_nearFriendlies
and
[position or object, radius, side, mode] call A3A_fnc_nearEnemies
```
position or object: determines the point around which should be searched
radius: determines the search radius around the point
side: the side of which we want to search friends or foes
mode; the mode in which should be searched, 0 return all, 1 return if

returns:
an array in mode 0, containing all friends/foes in the given search area
a boolean in mode 1, true if at least one friend/foe is in the given search area, false otherwise

That should replace all allunits searches and saves a few ms there and then. Every allUnit and distanceUnit has to be replaced for it

**Comment by Bob-Murphy** [comment]
@Spoffy @wurzel0701 
Will this be influenced by the AI system change?

**Comment by Spoffy** [comment]
No, it wouldn't. And I'm not sure this is something we can readily do.

**Comment by wurzel0701** [comment]
Spoffy is right, this has nothing to do with the AI system change, it is just for performance.
And yes we can do this whenever we want, I would say the earlier the better
